**Project I: PyBank**

In [1]:
from pathlib import Path
import csv

In [2]:
budgetpath = Path('Resources/budget_data.csv')
budgetpath

WindowsPath('Resources/budget_data.csv')

In [3]:
Date = []
Prof_Loss = []
dif_gain_loss = []
Total_Months = 0
Total = 0
avg_dif_gain_loss = 0


In [4]:
with open(budgetpath, 'r') as budgetfile:
    budgetreader = csv.reader(budgetfile, delimiter =',')
    budget_header = next(budgetreader)
    #print(f"Original Header: {budget_header}")
    
    # Adding new colunm for 'Monthly_Change' to the header
    #budget_header.append("Monthly_Change")
    #print(f"New Header: {budget_header}")
    
    for row in budgetreader:
            Date.append(row[0])
            Prof_Loss.append(float(row[1]))
                       
        
    print("**Financial Analysis**")
    print("-----------------------------------")
    print("Total Months:", len(Date) )
    print("Total Revenue: $", sum(Prof_Loss))
                
     
    for i in range(1,len(Prof_Loss)):
            dif_gain_loss.append(Prof_Loss[i] - Prof_Loss[i-1])   
            avg_dif_gain_loss = sum(dif_gain_loss)/len(dif_gain_loss)
        
            max_dif_gain_losse = max(dif_gain_loss)

            min_dif_gain_loss = min(dif_gain_loss)

            max_dif_gain_loss_Date = str(Date[dif_gain_loss.index(max(dif_gain_loss))+1])
            min_dif_gain_loss_Date = str(Date[dif_gain_loss.index(min(dif_gain_loss))+1])

    print("Avereage Change: $", round(avg_dif_gain_loss))
    print("Greatest Increase in Profits:", max_dif_gain_loss_Date,"($", max_dif_gain_losse,")")
    print("Greatest Decrease in Profits:", min_dif_gain_loss_Date,"($", min_dif_gain_loss,")")

            

**Financial Analysis**
-----------------------------------
Total Months: 86
Total Revenue: $ 38382578.0
Avereage Change: $ -2315
Greatest Increase in Profits: Feb-2012 ($ 1926159.0 )
Greatest Decrease in Profits: Sep-2013 ($ -2196167.0 )


 ###  Financial Analysis
 
  ----------------------------
  
  **Total Months: 86**
  
  **Total: $38382578**
  
  **Average Change: $-2315.12**
  
  **Greatest Increase in Profits: Feb-2012 ($1926159)**
  
  **Greatest Decrease in Profits: Sep-2013 ($-2196167)**
  